internal/daemon/controller/controller.go

```diff
@@ -489,19 +489,48 @@ func New(ctx context.Context, conf *Config) (*Controller, error) {
 		if c.conf.RawConfig.Controller.Help.ApiKey == "" {
 			return nil, fmt.Errorf(`api key is required with help model "gemini"`)
 		}
-		var opts []googleai.Option
+		opts := []googleai.Option{
+			googleai.WithAPIKey(c.conf.RawConfig.Controller.Help.ApiKey),
+			googleai.WithDefaultEmbeddingModel("models/text-embedding-004"),
+			googleai.WithDefaultModel("models/gemini-1.5-pro-latest"),
+		}
 		if c.conf.RawConfig.Controller.Help.Model != "" {
 			opts = append(opts, googleai.WithDefaultModel(c.conf.RawConfig.Controller.Help.Model))
 		}
 		if c.conf.RawConfig.Controller.Help.EmbeddingModel != "" {
 			opts = append(opts, googleai.WithDefaultEmbeddingModel(c.conf.RawConfig.Controller.Help.EmbeddingModel))
 		}
-		googleai, err := gemini.NewModel(ctx, c.conf.RawConfig.Controller.Help.ApiKey, opts...)
+		llm, err := googleai.New(ctx, opts...)
 		if err != nil {
 			return nil, fmt.Errorf("unable to initialize gemini model: %w", err)
 		}
-		c.llm = googleai
-		c.searcher, err = help.NewSearcher(ctx, eventLogger.Named("help"), dbase, dbase, googleai, c.conf.RawConfig.Controller.Help.NumHintDocuments)
+		c.llm = llm
+		c.searcher, err = help.NewSearcher(ctx, eventLogger.Named("help"), dbase, dbase, llm, c.conf.RawConfig.Controller.Help.NumHintDocuments)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create searcher: %w", err)
+		}
+		if err := c.searcher.CreateEmbeddings(ctx); err != nil {
+			return nil, fmt.Errorf("unable to create embeddings: %w", err)
+		}
+	case "openai":
+		if c.conf.RawConfig.Controller.Help.ApiKey == "" {
+			return nil, fmt.Errorf(`api key is required with help model "openai"`)
+		}
+		opts := []openai.Option{
+			openai.WithToken(c.conf.RawConfig.Controller.Help.ApiKey),
+		}
+		if c.conf.RawConfig.Controller.Help.Model != "" {
+			opts = append(opts, openai.WithModel(c.conf.RawConfig.Controller.Help.Model))
+		}
+		if c.conf.RawConfig.Controller.Help.EmbeddingModel != "" {
```

Comment:
What is the `EmbedddingModel` for openai? We also don't create embeddings when the llm is "openai"

Reply:
This relies on using the default embedding model for openai, I'm not sure which one it is, but I don't think it should matter so much.

We do create embeddings further down, it's just not showing up in this diff because the logic is repeated.